### PR TITLE
PLAT-124525: Fix when selecting `play` once and `next` several times

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Issue: A console error is shown when selecting `play` once and `next` several times on MediaPlayer in sampler.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Solution: Added asynchronous function for `play`.

### Links
[//]: # (Related issues, references)
PLAT-124525
https://github.com/enactjs/enact/tree/feature/PLAT-124525

### Comments

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]
I'm ok with this PR, as long as ui/Media cannot be changed. Otherwise, another PR in ui/Media is needed

File: MediaPlayer/MediaPlayer.js [hunk 1]
```diff
@@ -536,6 +536,18 @@ const MediaPlayerBehaviorDecorator = hoc((config, Wrapped) => { // eslint-disabl
...
 
+		asyncPlay = () => {
+			const playPromise = this.play();
+			if (playPromise) {
+				playPromise.then(() => {
+					this.play();
+				})
+					.catch(() => {
+						// Auto-play was prevented
+					});
```
Comment (ybsung) [inline MediaPlayer/MediaPlayer.js:547]: Please check indentation.

**Review by ybsung — APPROVED** [review]
LGTM

File: MediaPlayer/MediaPlayer.js [hunk 2]
```diff
@@ -536,6 +536,17 @@ const MediaPlayerBehaviorDecorator = hoc((config, Wrapped) => { // eslint-disabl
 			}
 		};
 
+		asyncPlay = () => {
+			const playPromise = this.play();
+			if (playPromise) {
+				playPromise.then(() => {
+					this.play();
```
Comment (jeonghee27) [inline MediaPlayer/MediaPlayer.js:543]: I wonder this line is needed.
I saw https://developers.google.com/web/updates/2017/06/play-request-was-interrupted and when i tried without this line it looks work.

If you don't need it, how about just modifying the play function without creating a separate asyncPlay function?

Comment (stanca-pop-lgp) [reply MediaPlayer/MediaPlayer.js]: You are right, I didn't need it. Thank you, I pushed the change.

**Pull request merged**